resilience4j-core/src/main/java/io/github/resilience4j/core/Registry.java

```diff
@@ -0,0 +1,59 @@
+/*
+ *
+ *  Copyright 2019 Mahmoud romeh
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *
+ */
+package io.github.resilience4j.core;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+
+/**
+ * root resilience4j registry to be used by resilience types registries for common functionality
+ */
+public interface Registry<Target, Config> {
```

Comment:
I think we should create a `AbstractRegistry` which implements the `Registry` interface.
It can be used by the other Registries as well.